### DISCUSSION TITLE
Flag on CLI airflow tasks test to attach debugger

### DISCUSSION THREAD

**Discussion by GuillaumeDesforges** [apache/airflow#22293] [Ideas]
**Description**

Any way to allow to attach a debugger when using `airflow tasks test`

**Use case / motivation**

Following [discussion](https://github.com/apache/airflow/discussions/14621):

I generate DAGs dynamically using a database.
I want to debug tasks.

The doc ([Debug Executor](https://airflow.apache.org/docs/apache-airflow/stable/executor/debug.html)) suggests to add

```python
if __name__ == '__main__':
  from airflow.utils.state import State
  dag.clear(dag_run_state=State.NONE)
  dag.run()
```
and run from IDE.

The above code snippet is something I'd rather avoid:
* it is hard to select the right dag and tasks when they are generated automatically
* it's a code artifact I'd rather avoid having, because everytime I test another dag, task or execution time, it creates git diffs

**Are you willing to submit a PR?**

No

**Related Issues**

Haven't seen any (?)

**Comment by ashb** [comment]
Are you aware of the the `dag.cli()` function? That might help you

**Comment by GuillaumeDesforges** [comment]
You seem to be referring to [this](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/models/index.html#airflow.models.DAG.cli)?

In order to use this command, I need to hard-write in my code which DAG to use, when my script generates a lot of them. I do not want to write which dag should be tested to the git history, when it changes frequently.
